modules/core/src/main/scala/fs2/kafka/KafkaConsumer.scala

```diff
@@ -534,7 +542,7 @@ private[kafka] object KafkaConsumer {
           streamId <- Stream.eval(streamIdRef.modify(n => (n + 1, n)))
           partitionStreamIdRef <- Stream.eval(Ref.of[F, Int](0))
           _ <- Stream.eval(initialEnqueue(streamId, partitions, partitionStreamIdRef))
```

Comment:
Btw, it's better to merge all these `eval`s into one, since each `eval` has a `race`  in it for interruption